### DISCUSSION TITLE
Schema design for a blog management system

### DISCUSSION THREAD

**Discussion by kanmaytacker** [kanmaytacker/fundamentals#18] [General]
What?
Create a schema diagram for a blog management system.

Why?
Practice. Important part of design.

How?
Fill [this](https://github.com/kanmaytacker/fundamentals/blob/sql-march-2023/database/practice/01-blog-schema-design.md) document.

**Comment by PrakashGatiyala** [comment]
# Schema 
-------------------------------------------------------------------------------------------------------------------
Users Table                                                                                                                               
id int primary key                                                                                                    
first_name, varchar(255) 
last_name, varchar(255)
mobile varchar(255)
email varchar(255) 
last_login_date date 
-------------------------------------------------------------------------------------------------------------------
## Users will have 1 to m relationship with Post Table 

Posts Table 
id int primary key
title varchar(255)
body varchar(255)
create_date date 
User_id int  ---> **Foreign Key to the Users table** 
status varchar(255) or ENUM type 
parent post varchar(255) 
comment_id int --> **Foreign key to comment table** 

-------------------------------------------------------------------------------------------------------------------
## Post and Tags will have m to m relationship so we will create middle table 
### This table will have m to 1 relationship with Post and Tags 
Post-Tags table 
tags_id int  --> **Foreign key to tags table** 
post_id  int --> **Foreign key to post table** 
tags_create_name  varchar(255) 
date_add date

-------------------------------------------------------------------------------------------------------------------
Tags Table 
id int primary key 
name varchar(255)  
description varchar(255)

-------------------------------------------------------------------------------------------------------------------
Comment Table
id int primary key 
user_id int --> **foreign key to users table**  

## Comment table will have 1 to 1 relationship with Post table 
## Users table will have 1 to m relationship with comment table 


-------------------------------------------------------------------------------------------------------------------

**Reply by kanmaytacker** [reply]
This looks great @PrakashGatiyala. 
Some things to improve - 
- Think about comment ID in post table? What is the cardinality there?
- Will a body of varchar (255) be sufficient for a post?

**Comment by garg04** [comment]
Here is the Schema link:- https://diagramplus.com/s/3vmgehagsdcl0rikch9myilsmuv5r6x1

**Reply by kanmaytacker** [reply]
This looks nice @garg04.
Some things to improve -

- You don't really need a second table for the parent post.
- A post can have more than one tag. How will you accommodate that?

**Comment by AbhinandanDas14** [comment]
file:///C:/Users/Abhinandan%20Das/Downloads/diagram%20(1).svg

**Reply by AbhinandanDas14** [reply]
[file:///C:/Users/Abhinandan%20Das/Downloads/diagram%20(2).svg](url)

**Comment by NitinDeogirikar** [comment]
https://diagramplus.com/auth/github/callback?code=fbdb214035ee3fcc6696&state=Bh1mrNiBW6SDtekR1AAbjK9phelIokHJcQJzts8OP6I%3D

**Comment by mohan-m2** [comment]
![image](https://user-images.githubusercontent.com/53464004/233796739-debab9c5-2079-44bd-966c-f724b3baa465.png)

**Comment by malayazone** [comment]
![diagram](https://user-images.githubusercontent.com/46066498/233796784-66ea40d7-9834-45a9-9876-22e4a12ee90a.png)

**Comment by ajaganbaskarz** [comment]
![image](https://user-images.githubusercontent.com/98110878/233796883-02f69107-329c-4348-9d50-d795a0de8379.png)
